go.mod

```diff
@@ -1,4 +1,6 @@
-go 1.21
+go 1.23
```

Comment:
It looks like this is causing a lot stuff to explode. I think will probs need to update a lot of files here to get things building. 

Do we need go1.23 in the celestia-core fork or was that just updated by happenstance? @tac0turtle

Reply:
nvm, looks like celestia-core and their sdk fork are already both using go1.23. think this is something we just have to deal with